packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -32,18 +62,11 @@ function computeSelfTime(span: Span, allSpans: Span[]): number {
   // We should've ended up with length 9-4=5, but we got 3.
   // To work around that, we multiply start/end times by 10 and subtract one from the end.
   // So instead of [1-10] we get [10-99]. This makes the intervals work like half-open.
-  const spanRange = new DRange(10 * span.startTime, 10 * (span.startTime + span.duration) - 1);
-  // Only keep CHILD_OF spans. FOLLOWS_FROM spans do not block the parent.
-  const children = allSpans.filter(
-    each =>
-      span.childSpanIds.includes(each.spanID) &&
-      each.references[0].spanID === span.spanID &&
-      each.references[0].refType === 'CHILD_OF'
+  if (!span.hasChildren) return span.duration;
+  const spanRange = new DRange(span.startTime, span.startTime + span.duration - 1).subtract(
```

Comment:
I don't think we can change the logic away from `*10`. DRange does not work correctly on the raw timestamps (see the comments above).

Side note: we need a test that illustrates that.

Reply:
I don't understand why the calculations need to change - the speed-up is coming from building a map ones, so why not change just that part and leave the calculations intact?

Reply:
That still doesn't answer my question why calculations need to change, it a PR that is about _efficiency_ of the code. 

If you want to dig down to the exact reasons for *10 logic, you can see the comment as well as the discussion on the original PR that introduced it. TLDR is that DRange does not understand open/close intervals, and subtraction works incorrectly for our purposes.